R/utils.R

```diff
@@ -140,3 +140,72 @@ resolve_modules_datanames <- function(modules, datanames, join_keys) {
     modules
   }
 }
+
+#' Check `datanames` in modules
+#'
+#' This function ensures specified `datanames` in modules match those in the data object,
+#' returning error messages or `TRUE` for successful validation.
+#'
+#' @param modules (`teal_modules`) object
+#' @param datanames (`character`) names of datasets available in the `data` object
+#'
+#' @return A `character(1)` containing error message or `TRUE` if validation passes.
+#' @keywords internal
+check_modules_datanames <- function(modules, datanames) {
+  recursive_check_datanames <- function(modules, datanames) {
+    # check teal_modules against datanames
+    if (inherits(modules, "teal_modules")) {
+      sapply(modules$children, function(module) recursive_check_datanames(module, datanames = datanames))
+    } else {
```

Comment:
just for safety -> add: else if inherits modules "teal_module" ... else stop(...)

Reply:
added assert on `modules` to be `teal_modules` - it is enough as modules can't contain anything else.